corehq/apps/callcenter/utils.py

```diff
@@ -189,18 +178,23 @@ def valid_element_name(name):
     return fields
 
 
-def _user_case_changed(case, case_type, close, fields, owner_id):
+def _get_changed_fields(case, case_type, owner_id, fields):
+    changed_fields = {}
+    if case.name != fields['name']:
+        changed_fields['name'] = fields['name']
+
+    if case.type != case_type:
+        fields['case_type'] = case_type
```

Comment:
This mutates an argument that was passed in, which could have unexpected side-effects elsewhere. I think you can change this to `changed_fields['case_type'] = case_type` and also change line 195 to `changed_fields.setdefault(field, value)` to avoid that (or use another approach if preferred).